### PR TITLE
statistics about wrong words

### PR DESCRIPTION
**Pull request by mirkoperillo**
Proposal for #26.
A `stats.json` file is created during the session  containing statistics about wrong words.

Starting the work on the GUI to consume the file

### CODE REVIEW

**Comment by mirkoperillo** [comment]
add simple pyqt5 GUI to read stats.json file

File: utils/data_handlers.py [hunk 1]
```diff
@@ -21,3 +21,16 @@ def save_typo_data(path, data):
...
+def save_stats_file(path, data):
+    """Save statistics about typos."""
+    with open(path, "w") as f:
+        f.write(
+            dumps({wrong_word: counter for wrong_word, counter in data.items()},
+                  indent=4,
+                  sort_keys=True))
+
+def open_stats_file(path):
+    """Open a file with statistics on wrong words and return the data as a defaultdict of sets."""
```
Comment (Mte90) [inline utils/data_handlers.py]: there is no check if the file exist

File: syntaxalert.py [hunk 2]
```diff
@@ -31,6 +32,17 @@ def loadWord(filename):
...
 files = {}
 
+wrong_word_counter = {}
+
+keyboard.start_recording()
+
+def mispell_callback():
+    recorded_words = keyboard.stop_recording()
+    recorded_words_list = list(keyboard.get_typed_strings(recorded_words))
+    wrong_word = recorded_words_list[0].split()[-1]
```
Comment (Mte90) [inline syntaxalert.py]: I got: 
```
  File "/usr/local/lib/python3.6/dist-packages/keyboard/_generic.py", line 22, in invoke_handlers
    if handler(event):
  File "/usr/local/lib/python3.6/dist-packages/keyboard/__init__.py", line 1106, in handler
    callback()
  File "/home/mte90/Desktop/Prog/syntaxalert/syntaxalert.py", line 42, in mispell_callback
    wrong_word = recorded_words_list[0].split()[-1]
IndexError: list index out of range
```

File: utils/data_handlers.py (hunk 1 above)
Comment (mirkoperillo) [reply utils/data_handlers.py]: yes you are right, if GUI starts and the file doesn't exist there is an exception.
I'll try to cover this scenario

File: syntaxalert.py (hunk 2 above)
Comment (mirkoperillo) [reply syntaxalert.py]: Exception is thrown every time you digit something or only some times ?

Comment (Mte90) [reply syntaxalert.py]: On the first word fixed I got that.

**Comment by mirkoperillo** [comment]
@Mte90 done

**Comment by Mte90** [comment]
immagine
Seems good but the same word is not aggregated and the json is the same.

**Comment by mirkoperillo** [comment]
I don't see the same word in the image you post, I see `volta` and `votla`. What I misunderstood ?

**Comment by Mte90** [comment]
you are right I misconfused -.-'

**Pull request merged**